src/test/java/org/openstreetmap/atlas/geography/PolygonTest.java

```diff
@@ -223,6 +224,196 @@ public void testNoAnglesMatchingOurCriteria()
         Assert.assertTrue(result.size() == 0);
     }
 
+    @Test
+    public void testNSidedness()
+    {
+        // Test point (has no sides)
```

Comment:
Really nice set of tests! 

- How about adding one more where the first one or two segments are points, and then the rest are not?
- How about one where the polygon is self intersecting? To make sure that the method does not throw exceptions?

Finally if you don't mind, it would be nice to break each use case in a different test method; that would make debugging easier down the line.